### DISCUSSION TITLE
how to use different mappings depending on file type?

### DISCUSSION THREAD

**Discussion by bigzhu** [NvChad/NvChad#2200] [Q&A]
Markdown files need mapping `F4` to `ObsidianOpen`, The python files need mapping `F4` to run code

**Comment by antoineco** [comment]
Vim — and Neovim by extension — has a super handy feature for that called `ftplugin`.

Here is [an example](https://github.com/antoineco/dotfiles/blob/5a3036c19f4d820c012da6f9b657d528894e0425/lvim/after/ftplugin/go.lua) of script that is executed whenever I open a file with the `go` filetype (notice the file name).

The example is from my LunarVim config and is not entirely adapted to NvChad. For NvChad you have to override/extend the `which-key` mappings [slightly differently](https://github.com/NvChad/NvChad/blob/720d71b546b8300bf3951c839f52db6cb83c3dc5/lua/plugins/init.lua#L261-L270). Or you can just create mappings outside of `which-key` using plain [`vim.api.set_buf_keymap`](https://neovim.io/doc/user/api.html#nvim_buf_set_keymap()), which is easier.

**Comment by bigzhu** [comment]
thanks
